docs/contributing/testing.md

```diff
@@ -0,0 +1,152 @@
+
+```bash
+node build/testing/sauce-labs/test-browsers.js alpha
+```
+
+[Beta](https://github.com/Microsoft/fast-dna/blob/master/build/testing/config-browsers.beta.js) Stable feature development
+
+```bash
+node build/testing/sauce-labs/test-browsers.js beta
+```
+
+[Release](https://github.com/Microsoft/fast-dna/blob/master/build/testing/config-browsers.release.js) Mature feature development
+
+```bash
+node build/testing/sauce-labs/test-browsers.js release
+```
+
+### Testing with Applitools
+
+[Applitools](https://eyes.applitools.com/) can be used to visual regression test UI changes on your own branches allow  to execute automated tests against Selenium WebDriver. Currently, ChromeDriver and FirefoxDriver are supported.
+
+Prior to beginning to make UI changes on a new branch it's recommended to run initially to create a baseline set of images.
+
+1. The first time the test is executed on a new branch, screenshots are created as 'undefined' status. 
+2. After the tests finish users should select all checkbox's and choose to save to generate the baseline image.
+3. Next perform UI code changes
+4. Rerun the first set of tests and for all snapshots that match the baseline they will turn green and show as 'passed' status. Any 'undefined' status would be additional UI changes added. Though to get into this scenario new test cases would be required in the `./run-msft-docs.js` file.  Any anomalies will be marked red as 'failed' status.
+5. Manually review all UI changes and if the changes are desired select and choose to save the new baseline. Anything, else should be changed and retested.
+
+```bash
+node build/testing/applitools/test-javascript.js
+```
+
+#### Visual Regression testing with Cypress SDK
+
+Contributors can perform functional user interface and experience testing using [Cypress](https://www.cypress.io/). Cypress is not based on Selenium WebDriver and instead is built on JavaScript enabling significant performance advantages.
+
+To validate Cypress is configured correctly.
+
+```bash
+npm run ui-tests:verify
+```
+
+Execute through Electron for real-time user interaction and debugging.
+
+```bash
+npm run ui-tests:open
+```
+
+Execute using the CLI.
+
+```bash
+npm run ui-tests:run
+```
+
+_When running as CLI, the results can be viewed as printed out to the command screen or in the cloud dashboard._
+
+## Setup reviews
+
+After development is complete it's a good idea to meet with key stakeholders for buy-in and acceptance. The following reviews can ensure everyone has time to evaluate and provide feedback prior to delivery.
+
++ Perform a design review
++ Perform a content review
++ Perform a functional review
++ Perform a performance review
++ Perform code reviews
+
+Use [ngrok](https://ngrok.com/) to securely tunnel your localhost to send test links, test on virtual machines, or test on other physical devices. When tunneling react pages, use the following format (replacing 7001 with your port):
```

Comment:
Ah so you do have this info in the documentation - I think they should be together in their own section... something like "#Physical testing using localhost" or something to that extent

Reply:
section removed per @nicholasrice